### PR TITLE
DAOS-12182 docker: Runing application in a docker container

### PR DESCRIPTION
**Pull request by knard-intel**
# Description

Defines how to use docker container for running an application accessing a DAOS file system.  the DAOS agent services should be run on the host node.

# Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

# Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by knard-intel** [comment]
My apologies, I did a rebase with my last minor modifications of the documentation: I forgot that the PR was not a draft any more.

File: utils/docker/cloud/.env [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# DAOS_CLIENT_GID="999"
+
+# XXX Arguments which must be updated if needed XXX
+DAOS_AGENT_RUNTIME_DIR="/var/run/daos_agent"
+
+# Arguments which should not be updated
+DAOS_DOCKER_IMAGE_TAG="rocky8.6"
+RHEL_BASE_IMAGE_NAME="rockylinux/rockylinux"
+RHEL_BASE_IMAGE_TAG="8.6"
+BUTST_CACHE=""
```
Comment (wiliamhuang) [inline utils/docker/cloud/.env:19]: Is it a typo here? BUST_CACHE="" ?

Comment (knard-intel) [reply utils/docker/cloud/.env:19]: Oups, indeed a typo error => going to fix it.

**Review by wiliamhuang — APPROVED** [review]

**Review by wiliamhuang — APPROVED** [review]

**Review by wiliamhuang — APPROVED** [review]

**Comment by knard-intel** [comment]
Update PR with a new commit allowing to mount dfuse mount point inside a DAOS client docker container.

**Review by wiliamhuang — APPROVED** [review]

**Comment by knard-intel** [comment]
@mark-olson please could you tell me if this PR is OK for you.
Thanks in advance.

**Review by mark-olson — APPROVED** [review]

**Comment by mark-olson** [comment]
LGTM

**Pull request merged**